src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
-                output_node = node_based_graph.GetTarget(in_edge);
-                intersection = intersection_generator(node, in_edge);
-            }
-            if (intersection.size() <= 2)
-            {
-                output_node = SPECIAL_NODEID;
-                intersection.clear();
-            }
-            return intersection;
-        };
+    if (!obvious)
+        return intersection;
 
-    const std::size_t obvious_turn_index = [&]() -> std::size_t {
-        const auto index = findObviousTurn(source_edge_id, intersection);
-        if (index != 0)
-            return index;
-        else if (intersection.size() == 3 && intersection[1].instruction.type == TurnType::Fork)
-        {
-            // Forks themselves do not contain a `obvious` turn index. If we look at a fork that has
-            // a one-sided sliproad, however, the non-sliproad can be considered `obvious`. Here we
-            // assume that this could be the case and check for a potential sliproad/non-sliproad
-            // situation.
-            NodeID intersection_node_one = SPECIAL_NODEID, intersection_node_two = SPECIAL_NODEID;
-            const auto intersection_following_index_one = findNextIntersectionForRoad(
-                intersection_node_id, intersection[1], intersection_node_one);
-            const auto intersection_following_index_two = findNextIntersectionForRoad(
-                intersection_node_id, intersection[2], intersection_node_two);
-            // in case of broken roads, we return
-            if (intersection_following_index_one.empty() ||
-                intersection_following_index_two.empty())
-                return 0;
-
-            // In case of loops at the end of the road, we will arrive back at the intersection
-            // itself. If that is the case, the road is obviously not a sliproad.
-
-            // a sliproad has to enter a road without choice
-            const auto couldBeSliproad = [](const Intersection &intersection) {
-                if (intersection.size() != 3)
-                    return false;
-                if ((intersection[1].entry_allowed && intersection[2].entry_allowed) ||
-                    intersection[0].entry_allowed)
-                    return false;
-                return true;
-            };
+    // Potential non-sliproad road (bc), leading to the intersection (c) the Sliproad (bd) shortcuts
+    const auto &next_road = intersection[*obvious];
 
-            if (couldBeSliproad(intersection_following_index_one) &&
-                intersection_node_id != intersection_node_two)
-                return 2;
-            else if (couldBeSliproad(intersection_following_index_two) &&
-                     intersection_node_id != intersection_node_one)
-                return 1;
-            else
-                return 0;
-        }
-        else
-            return 0;
-    }();
+    // The intersection `c` has to be reachable
+    if (!next_road.entry_allowed)
+    {
```

Comment:
Are we using `{}` for single line `if` branches, or not? I feel that this is often inconsistent in our code-base. Just 5 lines above you use the other style. I'd prefer not having the additional braces here.